### PR TITLE
[CPT-1075] Add tests to remove and clone query builder rules

### PR DESCRIPTION
**Pull request by toptalwadiibasmi**
[CPT-1075]

### Description

- This PR adds cypress tests for QueryBuilder removing and cloning rules.
- It adds test ids for clone and remove buttons.

### How to test

- Tests should be running

### Screenshots

None

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[CPT-1075]: https://toptal-core.atlassian.net/browse/CPT-1075?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: cypress/component/QueryBuilder/QueryBuilder.spec.tsx [hunk 1]
```diff
@@ -260,4 +271,42 @@ describe('QueryBuilder', () => {
       getGroupByDepth(4).should('be.visible').and('exist')
     })
   })
+
+  describe('adds, clones and removes rules', () => {
```
Comment (angelinastavniiciuc) [inline cypress/component/QueryBuilder/QueryBuilder.spec.tsx:275]: to keep description consistent with groups tests 
```suggestion
  describe('we manipulate query builder rules', () => {
```

**Review by angelinastavniiciuc — APPROVED** [review]
LGTM, minor suggestion

File: cypress/component/QueryBuilder/QueryBuilder.spec.tsx (hunk 1 above)
Comment (toptalwadiibasmi) [reply cypress/component/QueryBuilder/QueryBuilder.spec.tsx:275]: will do.

File: packages/picasso-query-builder/src/RemoveRuleButton/RemoveRuleButton.tsx [hunk 2]
```diff
@@ -7,11 +7,13 @@ export const RemoveRuleButton = ({
   handleOnClick,
   className,
   disabled,
+  context: { testIds },
```
Comment (dmaklygin) [inline packages/picasso-query-builder/src/RemoveRuleButton/RemoveRuleButton.tsx:10]: we usually use `testIds` prop: https://github.com/search?q=repo%3Atoptal%2Fpicasso%20testIds%3F%3A&type=code

I would recommend to use the same name

Comment (toptalwadiibasmi) [reply packages/picasso-query-builder/src/RemoveRuleButton/RemoveRuleButton.tsx:10]: This is a property that is inherited from the query builder library; it automatically prefills the component with a context property, in which we have previously put the testIds property. I am merely using it again in this component.

**Review by dmaklygin — APPROVED** [review]

**Comment by toptalwadiibasmi** [comment]
Is there a command I need to run to pass the Happo check?

**Pull request merged**